trino/base/hive-metastore-entrypoint.yaml

```diff
@@ -0,0 +1,120 @@
+        ALIAS="${PEM_FILE%.*}-$N"
+        cat $PEM_FILE |
+          awk "n==$N { print }; /END CERTIFICATE/ { n++ }" |
+          keytool -noprompt -import -trustcacerts \
+                  -alias $ALIAS -keystore $KEYSTORE -storepass $PASSWORD
+      done
+    }
+    set -e
+
+    # if the s3-compatible ca bundle is mounted in, add to the root Java truststore.
+    if [ -a /s3-compatible-ca/ca-bundle.crt ]; then
+      echo "Adding /s3-compatible-ca/ca-bundle.crt to $JAVA_HOME/lib/security/cacerts"
+      importCert /s3-compatible-ca/ca-bundle.crt changeit $JAVA_HOME/lib/security/cacerts
+    fi
+    # always add the openshift service-ca.crt if it exists
+    if [ -a /var/run/secrets/kubernetes.io/serviceaccount/service-ca.crt ]; then
+      echo "Adding /var/run/secrets/kubernetes.io/serviceaccount/service-ca.crt to $JAVA_HOME/lib/security/cacerts"
+      importCert /var/run/secrets/kubernetes.io/serviceaccount/service-ca.crt changeit $JAVA_HOME/lib/security/cacerts
+    fi
+
+    # add UID to /etc/passwd if missing
+    if ! whoami &> /dev/null; then
+        if test -w /etc/passwd || stat -c "%a" /etc/passwd | grep -qE '.[267].'; then
+            echo "Adding user ${USER_NAME:-hadoop} with current UID $(id -u) to /etc/passwd"
+            # Remove existing entry with user first.
+            # cannot use sed -i because we do not have permission to write new
+            # files into /etc
+            sed  "/${USER_NAME:-hadoop}:x/d" /etc/passwd > /tmp/passwd
+            # add our user with our current user ID into passwd
+            echo "${USER_NAME:-hadoop}:x:$(id -u):0:${USER_NAME:-hadoop} user:${HOME}:/sbin/nologin" >> /tmp/passwd
+            # overwrite existing contents with new contents (cannot replace the
+            # file due to permissions)
+            cat /tmp/passwd > /etc/passwd
+            rm /tmp/passwd
+        fi
+    fi
+
+    # symlink our configuration files to the correct location
+    if [ -f /hadoop-config/core-site.xml ]; then
+      cat /hadoop-config/core-site.xml | sed "s#XXX_S3ENDPOINT_XXX#${S3_BUCKET_NAME}/${S3_DATA_DIR}#" > /etc/hadoop/core-site.xml
+    else
+      echo "core-site.xml doesnt exist, skipping"
+    fi
+    if [ -f /hadoop-config/hdfs-site.xml ]; then
+      ln -s -f /hadoop-config/hdfs-site.xml /etc/hadoop/hdfs-site.xml
+    else
+      echo "hdfs-site.xml doesnt exist, skipping symlink"
+    fi
+
+    # insert S3 bucket URI from env vars
+    cat /hive-config/hive-site.xml | sed \
+      -e "s#XXX_S3ENDPOINT_URL_PREFIX_XXX#${S3_ENDPOINT_URL_PREFIX}#" \
+      -e "s#XXX_S3LOCATION_XXX#${S3_ENDPOINT}#" \
+      -e "s#XXX_S3ENDPOINT_XXX#${S3_BUCKET_NAME}/${S3_DATA_DIR}#" \
+      -e "s#XXX_DATABASE_USER_XXX#${DATABASE_USER}#" \
+      -e "s#XXX_DATABASE_PASSWORD_XXX#${DATABASE_PASSWORD}#" \
+      -e "s#XXX_DATABASE_CONNECT_URL_XXX#postgresql://${DATABASE_HOST}:${DATABASE_PORT}/${DATABASE_NAME}?sslmode=${DATABASE_SSLMODE}#" \
+    > $HIVE_HOME/conf/hive-site.xml
```

Comment:
Why don't you replace the values directly from kustomize vars in `hive-site.xml`?

If I understand this properly, you first load the values from a configmap/secret into a kustomize var, then you mount it to deployment's environment variables as a plaintext, literal value and then `sed` it into a configmap here? Wouldn't it be better to template the `hive-site.xml` directly via kustomize vars?

Or, if you really want to `sed` it during the entry script, mount the configmap/secret value directly to the environment?
https://github.com/opendatahub-io/odh-manifests/pull/373/files#diff-1781d34c25957ba53765bcf4e45efd79b6463410d5797be42c4cccb73a01a16eR32-R35